### PR TITLE
Add ScottoSlant Keyboard

### PR DESCRIPTION
**Pull request by joe-scotto**
Add ScottoSlant keyboard and default keymap. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/jscotto/scottoslant/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include QMK_KEYBOARD_H
+
+// Tap Dance declarations
```
Comment (drashna) [inline keyboards/handwired/jscotto/scottoslant/keymaps/default/keymap.c:20]: Could the tap dance stuff be moved to another keymap? 

We want to keep the default keymap fairly simple, as it's cloned when making a new keymap.  For newer users, complicated keymaps tend to cause more issues than they solve. 

That said, adding a "stock" type keymap with more features is fine.

Comment (joe-scotto) [reply keyboards/handwired/jscotto/scottoslant/keymaps/default/keymap.c:20]: Same as my other board... its for CMD+space mainly.

**Comment by joe-scotto** [comment]
@drashna Can you take a look at this when you get a chance? I think it's all set to merge unless you have other suggestions.

File: keyboards/handwired/jscotto/scottoslant/keymaps/default/keymap.c (hunk 1 above)
Comment (drashna) [reply keyboards/handwired/jscotto/scottoslant/keymaps/default/keymap.c:20]: https://docs.qmk.fm/#/pr_checklist?id=keyboard-prs

- default (and via) keymaps should be “pristine”
  - bare minimum to be used as a “clean slate” for another user to develop their own user-specific keymap
  - standard layouts preferred in these keymaps, if possible
  - default keymap should not enable VIA – the VIA integration documentation requires a keymap called via

Having the tap dance stuff would be fine for a "stock" type keymap, but the default kept and simple and pristine as possible.  

There are a number of reasons for this, but the biggest one is that the default keymap is copied when creating new keymaps and more advanced features can (and do) trip up a lot of new users.

**Pull request closed without merging**